### PR TITLE
Heap browser destination

### PR DESCRIPTION
**Pull request by GedRap**
This is the implementation of the Heap browser destination.

All the tests pass locally, I also tested the changes end-to-end using the local server. Details of testing and screenshots are available in PR descriptions here https://github.com/heap/action-destinations/pull/3 and here https://github.com/heap/action-destinations/pull/2 .

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/browser-destinations/src/destinations/heap/index.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+declare global {
+  interface Window {
+    heap: HeapApi
+  }
+}
+
+// Switch from unknown to the partner SDK client types
+export const destination: BrowserDestinationDefinition<Settings, HeapApi> = {
+  name: 'Heap',
```
Comment (pooyaj) [inline packages/browser-destinations/src/destinations/heap/index.ts:17]: Could we change the name to: 
* `Heap Web (Actions)` if we anticipate there will be a cloud mode destination 
* or `Heap (Actions)` if we think that there will be only a browser destination for the foreseeable future 
The reason for this is that we already have a classic "Heap" destination in the catalog

File: packages/browser-destinations/src/destinations/heap/index.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  slug: 'heap',
```
Comment (pooyaj) [inline packages/browser-destinations/src/destinations/heap/index.ts:18]: let's change the slug to `actions-heap-web`

**Review by pooyaj — COMMENTED** [review]
The logic looks great. Couple of comments on naming, otherwise, looks good to go

File: packages/browser-destinations/src/destinations/heap/trackEvent/__tests__/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  createMocekdHeapJsSdk,
+  HEAP_TEST_ENV_ID,
+  mockHeapJsHttpRequest,
+  trackEventSubscription
+} from '../../test-utilities'
+
+describe('#trackEvent', () => {
+  it('sends events to heap', async () => {
+    mockHeapJsHttpRequest()
+    window.heap = createMocekdHeapJsSdk()
```
Comment (silesky) [inline packages/browser-destinations/src/destinations/heap/trackEvent/__tests__/index.test.ts:13]: createMocekdHeapJsSdk -> createMockedHeapJsSdk =)

Comment (silesky) [inline packages/browser-destinations/src/destinations/heap/trackEvent/__tests__/index.test.ts:13]: createMocekdHeapJsSdk -> createMockedHeapJsSdk =)

**Review by pooyaj — APPROVED** [review]

**Review by silesky — APPROVED** [review]
LGTM

File: packages/browser-destinations/src/destinations/heap/types.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+export type UserConfig = {
```
[the remark this answers is not in the document]
Comment (GedRap) [reply packages/browser-destinations/src/destinations/heap/types.ts:1]: Good point! No particular reason behind this, to be honest. So I'm going to convert `HeapApi` to a type.

[the remark this answers is not in the document]
Comment (silesky) [reply packages/browser-destinations/src/destinations/heap/types.ts:1]: Thank you!

**Pull request merged**